source/gfoidl.Stochastics/Partitioners/TrapezeWorkloadPartitioner.cs

```diff
@@ -0,0 +1,127 @@
+     * areas.
+     * 
+     *              A
+     *              |\
+     *              |  \
+     *              |   |\
+     *              |   |  \ X
+     *              |   |    \
+     *              |   |    | \
+     *              |   |    |   \ B
+     *              |   |    |     \
+     *              | A1| A2 | A3  |
+     *              |   |    |     |
+     *              |___|____|_____| 
+     *              0        n     N
+     *          
+     *  (0, A) is the load at the first iteration
+     *  (N, B) is the load at the last iteration
+     *  
+     *  The constant part with load to B can be eliminated, so the partitioning can be done on a triangle.
+     *  Thus the calculations are (greatly) simplified.
+     */
+    internal class TrapezeWorkloadPartitioner : WorkloadPartitioner
+    {
+        private readonly double _lambda;
+        //---------------------------------------------------------------------
+        public TrapezeWorkloadPartitioner(
+            int    size,
+            double lambda,
+            int?   partitionCount = null)
+            : base(size, partitionCount)
+            => _lambda = lambda;
+        //---------------------------------------------------------------------
+        public TrapezeWorkloadPartitioner(
+            int    size,
+            double loadFactorAtStart,
+            double loadFactorAtEnd,
+            int?   partitionCount = null)
+            : this(size, (loadFactorAtEnd - loadFactorAtStart) / size, partitionCount)
+        { }
+        //---------------------------------------------------------------------
+        protected override IEnumerable<KeyValuePair<long, Range>> GetOrderableDynamicPartitions(int partitionCount)
+            => new TrapezeWorkloadPartitions(_size, _lambda, partitionCount);
+        //---------------------------------------------------------------------
+        private class TrapezeWorkloadPartitions : Partitions
+        {
+            private readonly Range[] _partitions;
+            //-----------------------------------------------------------------
+            public TrapezeWorkloadPartitions(
+                int    size,
+                double lambda,
+                int    partitionCount)
+                : base(size, partitionCount)
+            {
+                _partitions = lambda == 0
+                    ? Rectangle()
+                    : Triangle();
+                //-------------------------------------------------------------
+                Range[] Triangle()
+                {
+                    bool increasing = true;
+                    if (lambda < 0)
+                    {
+                        increasing = false;
+                        lambda     = -lambda;
+                    }
+
+                    var partitions       = new Range[partitionCount];
+                    double triangleArea  = size * size * lambda * 0.5;
```

Comment:
`size` is int, so `size * size` may overflow, because the compiler makes a double first when multiplyed by `lambda` 😞

Reply:
Rearranging to 
```c#
double triangleArea  = lambda * 0.5 * size * size;
```
works, but to be save there should be an explicit cast to `double`. One doesn't know if the behavior of the compiler changes, and there there will be subtle bugs due the use of _unsafe_ that result in AccessViolations.